use-cases/bed-allocation/src/main/java/org/acme/bedallocation/domain/BedDesignation.java

```diff
@@ -55,7 +55,7 @@ public Integer getPatientPreferredMaximumRoomCapacity() {
     }
 
     @JsonIgnore
-    public Specialism getSpecialism() {
+    public String getSpecialism() {
```

Comment:
I'm not sure how I feel about this. Two aspects:

- `Specialism` is a word I've never seen used anywhere else other than this quickstart. What is it supposed to mean, and can we find a better one?
- To me, `String` is not a data type. It carries no meaning, you cannot effectively use it in constraints. I consider this to be an anti-pattern, and I'd rather not teach our users to do this. I prefer the original approach, where this had a specific type for itself.

This has been a point where Geoffrey and I've disagreed over the years. If the `specialism` is never ever used inside the constraints, or rather used through `BedSpecialism` et al., then I can live with this. But I still don't like the name.

Reply:
I don't have a strong opinion on that. 

> I'm not sure how I feel about this. Two aspects:
> 
> * `Specialism` is a word I've never seen used anywhere else other than this quickstart. What is it supposed to mean, and can we find a better one?

During some quick research, I discovered more than one work that utilizes the term `Specialty`. What do you think?

> * To me, `String` is not a data type. It carries no meaning, you cannot effectively use it in constraints. I consider this to be an anti-pattern, and I'd rather not teach our users to do this. I prefer the original approach, where this had a specific type for itself.
> 

I see your point. I think the objective is to simplify the design of the domain models. Regardless, other quickstart modules use the same approach, e.g., `employee scheduling`.

Reply:
Just like the school timetabling quickstarts, that uses a String for Teacher etc, it makes sense for simplicity's sake to use a String for Speciality (or Specialism) and Equipment here. Keep it simple.

Reply:
Once a data type has a property beyond its name, then it should no longer be String.